test/module/irohad/validation/query_execution.cpp

```diff
@@ -474,14 +532,19 @@ TEST_F(GetAccountAssetsTest, DifferentDomainAccountInValidCase) {
       .WillOnce(Return(admin_roles));
   EXPECT_CALL(*wsv_query, getRolePermissions(admin_role))
       .WillOnce(Return(role_permissions));
-  EXPECT_CALL(*wsv_query,
-              hasAccountGrantablePermission(
-                  admin_id, get_account_assets->account_id, can_get_my_acc_ast))
+  EXPECT_CALL(
+      *wsv_query,
+      hasAccountGrantablePermission(admin_id, "test@test2", can_get_my_acc_ast))
       .WillOnce(Return(false));
 
-  auto response = validateAndExecute();
-  auto cast_resp = std::static_pointer_cast<ErrorResponse>(response);
-  ASSERT_EQ(cast_resp->reason, ErrorResponse::STATEFUL_INVALID);
+  auto response = validateAndExecute(query);
+
+  auto cast_resp = boost::get<w<shared_model::interface::ErrorQueryResponse>>(
+      response->get());
+
+  ASSERT_NO_THROW(
```

Comment:
`ASSERT_NO_THROW` with a cast of the variant inside is very common. Maybe better to move it into own function?

Reply:
It is 2 lines of code, the function would have more, and we have to pass the response as parameter and the error as template, so it is an overkill